react/src/components/ShellScriptEditModal.tsx

```diff
@@ -0,0 +1,181 @@
+          onRequestClose();
+        })
+        .catch((error) => {
+          message.error(error.message);
+          console.error(error);
+        });
+    }
+
+    if (shellInfo === 'userconfig') {
+      const existValidator = _.find(userConfigScript, { path: rcfileNames });
+      if (existValidator) {
+        (
+          baiRequestWithPromise({
+            method: 'PATCH',
+            url: '/user-config/dotfiles',
+            body: {
+              data: script,
+              path: rcfileNames,
+              permission: '644',
+            },
+          }) as Promise<object>
+        )
+          .then(() => {
+            message.success(t('usersettings.BootstrapScriptUpdated'));
+            onRequestClose();
+          })
+          .catch((error) => {
+            message.error(error.message);
+            console.error(error);
+          });
+      } else {
+        (
+          baiRequestWithPromise({
+            method: 'POST',
+            url: '/user-config/dotfiles',
+            body: {
+              path: rcfileNames,
+              data: script,
+              permission: '644',
+            },
+          }) as Promise<object>
+        )
+          .then(() => {
+            message.success(t('usersettings.BootstrapScriptUpdated'));
+            onRequestClose();
+          })
+          .catch((error) => {
+            message.error(error.message);
+            console.error(error);
+          });
+      }
+    }
+  };
+
+  return (
+    <BAIModal
+      width={800}
+      title={t('usersettings.EditBootstrapScript')}
+      onCancel={() => onRequestClose()}
+      onOk={() => saveBootstrapScript()}
+      afterOpenChange={(open) => {
+        open && initialFetch();
+      }}
+      okText={t('button.Save')}
+      destroyOnClose
+      {...modalProps}
+    >
+      <Flex direction="column" align="stretch" gap={'sm'}>
+        {shellInfo === 'userconfig' && (
+          <Select
+            onClear={() => setRcfileNames('.bashrc')}
+            defaultValue={'.bashrc'}
+            onChange={(value) => {
+              const selectedScript = _.find(userConfigScript, { path: value });
+              codeMirrorRef?.current?.setValue(selectedScript?.data || '');
+              setRcfileNames(value);
+            }}
+            options={[
+              { value: '.bashrc', label: '.bashrc' },
+              { value: '.zshrc', label: '.zshrc' },
+              { value: '.tmux.conf.local', label: '.tmux.conf.local' },
+              { value: '.vimrc', label: '.vimrc' },
+              { value: '.Renviron', label: '.Renviron' },
+            ]}
+            style={{ width: '200px' }}
+          />
+        )}
+        <Typography.Text>
+          {t('usersettings.BootstrapScriptDescription')}
+        </Typography.Text>
+        {/* @ts-ignore */}
+        <lablup-codemirror
+          ref={codeMirrorRef}
+          id="bootstrap-editor"
+          mode="shell"
+        />
```

Comment:
Please avoid using lablup-codemirror in React as it is only used for the classic session launcher. It will be deprecated soon. Let's use a code editor in the React components instead.
Please review this module. If there is a better component available, please inform me.
https://www.npmjs.com/package/react-codemirror2

Reply:
There is an issue with the rendering of react-codemirror2. However, I think I can solve this via CSS.

[ref](https://stackoverflow.com/questions/73518247/react-codemirror2-creating-two-editors)

Reply:
I found other issues with using code-mirror inside other components.
[ref](https://github.com/scniro/react-codemirror2/issues/83)

Reply:
I'm getting too many bugs to use controlled code mirror. I've tried my best to work around them, but it looks like I'm better off finding another way.

Reply:
https://www.npmjs.com/package/react-codemirror2 This library has support issues with react 18 and 19...

The library below seems to work well in react 18 and above. I will try working with that library again.

https://uiwjs.github.io/react-codemirror/